### ISSUE TITLE
HtmlValidation often fails on Orchard Sites when doing UITests.

### ISSUE THREAD

**Issue by Psichorex** [OrchardCMS/OrchardCore#12271]
### Describe the bug
When UITesting an Orchard App Html Validation usually fails the UI Test.
However this is only happening when you are using RelativeUrl navigation.
Basically using Lombiq.Tests.UI extensions `GoToDashboardAsync()` won't fail but `GoToRelativeUrlAsync("/Admin")` will do due to html validation errors.
### To Reproduce
1. Go to any UI test of you app and replace `GoToExactPage()` with `GoToRelativeUrl("/uri")`
2. Rerun the UI test. Now it should fail due to various bad html syntaxes like <language> attribute is required etc.
3. This is only problematic mostly on sites that uses custom admin prefix as for those navigations `GoToRelativeUrl()` is utilized mostly.

### Expected behavior
Navigating relatively with URLs shouldn't break a UITest. I think that `GoToDashboardAsync()` and others are not failing because they don't navigate with  GoToRelativeUrl but with `context.Scope.AtataContext.Go.To<TPage().` Methods navigating like this don't trip html validation.

### Screenshots
Here you can see that Admin Prefix was altered thus the navigations will use relative urls and the test is failing due to html validation.
![image](https://user-images.githubusercontent.com/92299130/186876496-8dfa17cd-9e60-44b9-86cb-e8a17edbbd3f.png)

**Comment by Piedone** [comment]
Note that people here don't know about what you do with UI testing. Focus on the what the issue is in Orchard, exactly (elaborate on what you show in the screenshot, everything else is not interesting for others here).

**Comment by Psichorex** [comment]
@Piedone Will elaborate on this more today. FYI I think this error is coming from our extensions. I think the old navigators that are for default prefix disable these html validations while GoToRelativeUrl doesn't. I will update here asap.

**Comment by hishamco** [comment]
I'm suggesting to move this into Lombiq UI Testing ToolBox repo, unless there's an issue related to Orchard Core

**Comment by Psichorex** [comment]
@hishamco Well there are 2 approaches for this issue.
1) Orchard Core HTML templates are not following HTML validation rules thus this problem is related to Orchard Core.
2) We don't care about  OC not following these and Lombiq UI Testing Toolbox should just neglect these validations.

It's up to the community which route shall we take.

**Comment by Piedone** [comment]
It's an issue with the validity of the HTML of the Orchard admin. Please edit the issue so it's actually about that, not about UI tests, since that's not relvant.

**Comment by hishamco** [comment]
> Orchard Core HTML templates are not following HTML validation rules thus this problem is related to Orchard Core

@Psichorex could you please elaborate which rules are you talking about, or you mean all of the HTML validation rules?

**Comment by Psichorex** [comment]
@hishamco There are too many to name. Probably not all but several rules were neglected when constructing the HTML of Orchard Admin. E.g. html-has-lang , rules about `<button>` elements and so on. I can't name them all. Basically the Orchard Admin pages' html should be presented to an  html validator and that would be able to filter out all the missing stuff.

**Comment by hishamco** [comment]
We could try to use an HTML validation service, unless @Piedone prefer another solution because already he working on Lombiq UI Testing ToolBox

**Comment by Skrypt** [comment]
Here is one that should be used : https://accessibilityinsights.io/docs/web/overview/

**Comment by hishamco** [comment]
Can we setup a tenant on try.orchardcore.com to test the admin site

**Comment by Piedone** [comment]
@Psichorex it's exactly three issues with a button on the admin ;). Please just add the output of the HTML validator to the issue.

**Comment by Psichorex** [comment]
@Piedone Here it's 3. But when I was adding UI testing to OC Commerce the BasicFeaturesTests produces much more similar ones.

**Comment by Piedone** [comment]
Where's "here"? All I'm asking is, copy the output of the HTML validator, something that UI tests produce in a text file, into the issue. The screenshot lacks a lot of information.

**Comment by Piedone** [comment]
So, could you please add the necessary information?

**Comment by Psichorex** [comment]
I will check OC's HTML as soon as I can and everything will be uploaded here.

**Comment by hishamco** [comment]
As @Skrypt said they are for accessibility, BTW I already started to enhance some of these but I stopped because I'm engaged with many important things. We could file an issue and submit a PR later, for now I suggest to fix HTML validations (as @Piedone said there're 3)

**Comment by Psichorex** [comment]
[HtmlValidationReport.txt](https://github.com/OrchardCMS/OrchardCore/files/9619702/HtmlValidationReport.txt)
Here is the HTML validation report from the UI test.
I also copied some of the OC admin dashboard's HTML into a REAL html validator like this: https://validator.w3.org/
And I have seen more errors there than in the UI test's report.
This is the /admin
![image](https://user-images.githubusercontent.com/92299130/191591320-f6a63f9a-63c0-4e99-87d0-2018240bc106.png)
This is /Admin/ContentItems
![image](https://user-images.githubusercontent.com/92299130/191591481-d479deab-80f8-411c-b721-2a07e72ad0ec.png)
You can see it found 25 we can subtrack 1-2 because it was missing !DOCTYPE that I didn't copy over but it will still be 20+

**Comment by Skrypt** [comment]
Some of them seem related to the Admin menu. I may take a look at that part when I get time but not now.

**Comment by Psichorex** [comment]
> Some of them seem related to the Admin menu. I may take a look at that part when I get time but not now.

Literally all of them are in the Admin Dashboard's html as I observed.

**Comment by Skrypt** [comment]
Yes, I remember that one "number 23 to 25" but I remember that we decided to keep it as it is for now. So, we just need someone that has time to fix all these small issues one by one.

**Comment by hishamco** [comment]
I might handle them after finishing some PRs
